src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -386,11 +386,31 @@ private void OnHostExited(HostProviderEventArgs e)
             }
         }
 
+        private void WarnIfDebugEnvironment()
+        {
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("VSTEST_HOST_NATIVE_DEBUG")))
```

Comment:
This should check if the variable is set to 1, the same way the internal check does it.